api/internal/handler/route/route_test.go

```diff
@@ -1349,3 +1349,146 @@ func Test_Route_With_Script(t *testing.T) {
 	_, err = handler.BatchDelete(ctx)
 	assert.Nil(t, err)
 }
+
+func Test_Route_With_Script_Luacode(t *testing.T) {
+	// init
+	err := storage.InitETCDClient(conf.ETCDConfig)
+	assert.Nil(t, err)
+	err = store.InitStores()
+	assert.Nil(t, err)
+
+	handler := &Handler{
+		routeStore:    store.GetStore(store.HubKeyRoute),
+		svcStore:      store.GetStore(store.HubKeyService),
+		upstreamStore: store.GetStore(store.HubKeyUpstream),
+		scriptStore:   store.GetStore(store.HubKeyScript),
+	}
+	assert.NotNil(t, handler)
+
+	// create with script of valid lua syntax
+	ctx := droplet.NewContext()
+	route := &entity.Route{}
+	reqBody := `{
+		  "id": "1",
+		  "uri": "/index.html",
+		  "upstream": {
+		      "type": "roundrobin",
+		      "nodes": [{
+		          "host": "www.a.com",
+		          "port": 80,
+		          "weight": 1
+		      }]
+		  },
+		  "script": "local _M = {} \n function _M.access(api_ctx) \n ngx.log(ngx.INFO,\"hit access phase\") \n end \nreturn _M"
+		}`
```

Comment:
Style: JSON format looks not very good.
I can see different  JSON format in your code, maybe you should format them like

https://github.com/apache/apisix-dashboard/pull/1289/files#diff-69dbad33b4168d98e52aecaff8a2997d5101e9936a45adf502a3cde9cb9f1763R1406-R1418

Reply:
@starsz PR renewed. The style will be fine if we only use  `table` to indent the JSON defined as a string type. Thanks for advice.